src/swap/SwapAmountInput.tsx

```diff
@@ -57,40 +58,53 @@ const SwapAmountInput = ({
     }
   }
 
+  const showInputLoader = loading && !inputValue
+
   return (
     <View style={[styles.container, style]} testID="SwapAmountInput">
       <Text style={styles.label}>{label}</Text>
       <View style={styles.contentContainer}>
-        <TextInput
-          forwardedRef={textInputRef}
-          onChangeText={(value) => {
-            handleSetStartPosition(undefined)
-            onInputChange(value)
-          }}
-          value={inputValue || undefined}
-          placeholder="0"
-          style={styles.input}
-          keyboardType="numeric"
-          autoFocus={autoFocus}
-          // unset lineHeight to allow ellipsis on long inputs on iOS
-          inputStyle={[
-            { lineHeight: undefined },
-            inputError ? styles.inputError : {},
-            loading ? styles.inputLoading : {},
-          ]}
-          testID="SwapAmountInput/Input"
-          onBlur={() => {
-            handleSetStartPosition(0)
-          }}
-          onFocus={() => {
-            handleSetStartPosition(inputValue?.length ?? 0)
-          }}
-          selection={
-            Platform.OS === 'android' && typeof startPosition === 'number'
-              ? { start: startPosition }
-              : undefined
-          }
-        />
+        {showInputLoader ? (
+          <View style={styles.loadingContainer}>
+            <ActivityIndicator
+              size="small"
+              color={Colors.greenBrand}
+              testID="SwapAmountInput/Loader"
+            />
+          </View>
```

Comment:
nit: no risk of things moving slightly while loading because the height is different?

Reply:
ah yes, one way to avoid the jump is to keep rendering the normal `TextInput`, but with `opacity: 0` while loading, and render the loading container absolutely on top.

Reply:
ah, that is so clever! i didn't think of this. i like it, i updated the component to do this